planner/core/find_best_task.go

```diff
@@ -469,6 +469,30 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty) (t task, err
 				p: dual,
 			}, nil
 		}
+		if !ds.isPartition && len(path.Ranges) > 0 {
+			allRangeIsPoint := true
+			for _, ran := range path.Ranges {
+				if !ran.IsPoint(ds.ctx.GetSessionVars().StmtCtx) {
+					allRangeIsPoint = false
+					break
+				}
+			}
+			if allRangeIsPoint {
+				var pointGetTask task
+				var err error
+				if len(path.Ranges) == 1 {
+					pointGetTask, err = ds.convertToPointGet(prop, candidate)
+				} else {
+					pointGetTask, err = ds.convertToBatchPointGet(prop, candidate)
+				}
+				if err != nil {
+					return nil, err
+				}
```

Comment:
Check if the returned task is `invalidTask`?

Reply:
The cost of `invalidTask` is always the max float64. Never be choosen.